lib/chef/provider/service/macosx.rb

```diff
@@ -77,6 +95,11 @@ def define_resource_requirements
               @current_resource.running(false)
             end
           end
+
+          requirements.assert(:all_actions) do |a|
+            a.assertion { @plist.nil? ? true : ::File.exists?(@plist) }
+            a.failure_message Chef::Exceptions::Service, "#{@plist} does not exist"
+          end
```

Comment:
Sorry, I didn't post more output for brevity. I wasn't testing why-run. Full output here: https://gist.github.com/btm/75b0975832809c35a912

This is a problem that seems to be cropping up more and more. It's the same thing we were talking about in the yum_resource recently: https://github.com/chef/chef/pull/3065#discussion_r26357705. I don't think I'm missing something here.

As @tyler-ball mentioned here, define_resource_requirements seems like the right place to, uh, define requirements of the resource. And thus the right place to check that a file exists. But if we're not in why-run mode, Provider.run_action calls load_current_resource, then define_resource_requirements, and then process_resource_requirements. In this provider load_current_resource examines the plist file before the guard in define_resource_requirement has been evaluated.

So define_resource_requirements is mostly for the benefit of why-run. If we want to raise a sane error when the user specifies a plist file that doesn't exist, e.g. a typo in a file name,, we have to do it earlier in a load_current_resource.

@mikedodge04 you didn't see this error in the example you tried with cookbook_file + macosx_service